packages/fleather/lib/src/widgets/editor_input_client_mixin.dart

```diff
@@ -57,7 +57,7 @@ mixin RawEditorStateTextInputClientMixin on EditorState
           inputType: TextInputType.multiline,
           readOnly: widget.readOnly,
           obscureText: false,
-          autocorrect: false,
+          autocorrect: widget.autocorrect,
```

Comment:
Could you please see if you can add a test for this? You can check https://github.com/fleather-editor/fleather/blob/34b56e7f69cb76f5dfaf1b40219c010a50aa638d/packages/fleather/test/widgets/editor_input_client_mixin_test.dart#L11-L24 for writing the test.

Reply:
Did you have in mind a test something like this?

```dart
testWidgets('enters text that should be auto corrected', (tester) async {
  bind(tester);
  final document = ParchmentDocument();
  final editor = EditorSandBox(tester: tester, document: document);
  await editor.pump();
  await editor.tap();
  tester.binding.scheduleWarmUpFrame();
  await tester.pumpAndSettle();
  await editor.enterText(const TextEditingValue(text: 'hello wolrd '));
  await tester.pumpAndSettle(throttleDuration);
  final text = editor.document.toPlainText();
  expect(
    text,
    'hello world',
  );
});
```

I have yet to find a find a way to mock the auto correction service, but will research it further. I would have expected this to work, but it is never called.

```dart
void bind(WidgetTester tester) {
   // ...
  tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(
      SystemChannels.spellCheck, (MethodCall methodCall) async {
    print("SystemChannels.spellCheck ${methodCall.method}");
    return null;
  });
}
```

Reply:
Just testing that we are enabling/disabling the autocorrect when creating `TextInputConnection` would be enough. I believe you can check for `TextInput.setClient` method call using `setMockMethodCallHandler`. If you want, I can push a commit for the test.

Reply:
Make sense! Added now. Looks good?